frontend/internal-packages/agent/src/langchain/agents/pmAgent/agent.ts

```diff
@@ -15,46 +16,73 @@ export const requirementsAnalysisSchema = v.object({
   nonFunctionalRequirements: v.record(v.string(), v.array(v.string())),
 })
 
-export class PMAgent implements ChatAgent {
-  private model: ChatOpenAI
+// TODO: Add review schema
+const reviewResponseSchema = v.object({
+  review: v.string(),
+})
+
+type AnalysisResponse = v.InferOutput<typeof requirementsAnalysisSchema>
+type ReviewResponse = v.InferOutput<typeof reviewResponseSchema>
+
+// Union type for all possible responses
+type PMAgentResponse = AnalysisResponse | ReviewResponse
+
+export class PMAgent implements ChatAgent<PMAgentResponse> {
+  private analysisModel: ReturnType<ChatOpenAI['withStructuredOutput']>
+  private reviewModel: ReturnType<ChatOpenAI['withStructuredOutput']>
 
   constructor() {
-    this.model = new ChatOpenAI({
+    const baseModel = new ChatOpenAI({
       model: 'o3',
       callbacks: [createLangfuseHandler()],
     })
+
+    // Convert valibot schemas to JSON Schema and bind to models
+    const analysisJsonSchema = toJsonSchema(requirementsAnalysisSchema, {
+      errorMode: 'ignore',
```

Comment:
Since this is a general schema definition, this specification should not be necessary in this case.

Reply:
I had missed it. Corrected.🙏
https://github.com/liam-hq/liam/pull/2134/commits/f4dfa101e999af92caa17e8087133d0d7419fafa